static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -173,27 +172,23 @@ function useReplayData({
     const finishedAtClone = new Date(replayRecord.finished_at);
     finishedAtClone.setSeconds(finishedAtClone.getSeconds() + 1);
 
-    const chunks = chunk(replayRecord.error_ids, errorsPerPage);
-    await Promise.allSettled(
-      chunks.map(errorIds => {
-        const promise = api.requestPromise(
-          `/organizations/${orgSlug}/replays-events-meta/`,
-          {
-            query: {
-              start: replayRecord.started_at.toISOString(),
-              end: finishedAtClone.toISOString(),
-              query: `id:[${String(errorIds)}]`,
-            },
-          }
-        );
-        promise.then(response => {
-          setErrors(prev => (prev ?? []).concat(response.data || []));
-        });
-        return promise;
-      })
-    );
+    const paginatedErrors = fetchPaginatedReplayErrors(api, {
+      orgSlug,
+      replayId: replayRecord.id,
+      start: replayRecord.started_at,
+      end: finishedAtClone,
+      limit: errorsPerPage,
+    });
+    let done = false;
+    while (!done) {
+      const pagedResults = await paginatedErrors.next();
+      done = pagedResults.done ?? false;
+      if (pagedResults.value && !done) {
+        setErrors(prev => [...prev, ...pagedResults.value]);
+      }
+    }
```

Comment:
I think we should be able to treat the generator like an iterator and not have to manually do the `let done` stuff

something like this should work, but i'm rusty
```
const paginatedErrorsIterator = fetchPaginatedReplayErrors(api, {...});

for await (const pagedResults of paginatedErrorsIterator) {
  setErrors(prev => [...prev, ...pagedResults];
}
```

I had to look it up again at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator

Reply:
super dope, completely forgot about `for await` the time has finally come to use it